BrowserKit/Sources/Shared/TimeConstants.swift

```diff
@@ -238,3 +238,14 @@ public func millisecondsToSeconds(_ input: Timestamp) -> UInt64 {
     let val = input / 1000
     return val
 }
+
+public protocol DateProvider {
```

Comment:
Good idea! One nit, could we put this in its own file maybe?

Reply:
will do! I'll wait for @Cramsden's initial review 👍 on the draft + will update the tests code